lib/showcase_widget.dart

```diff
@@ -29,8 +29,10 @@ import 'package:flutter/material.dart';
 class ShowCaseWidget extends StatefulWidget {
   final Builder builder;
   final VoidCallback onFinish;
+  final Function onStart;
+  final Function onComplete;
```

Comment:
The type `Function` is too generic. Perhaps we should make it statically type safe and restrictive to our use case.

e.g. `Function(int, GlobalKey)` or `typedef ShowCaseEventListener = void Function(int, GlobalKey);`

I prefer the later one. though I am okay with either way.